lua/telescope/pickers/layout_strategies.lua

```diff
@@ -476,25 +483,38 @@ layout_strategies.center = make_documented_layout(
       error(string.format("Unknown prompt_position: %s\n%s", self.window.prompt_position, vim.inspect(layout_config)))
     end
 
-    if not layout_config.mirror then
-      preview.line = 1 + bs
-      preview.height = topline - (2 * bs + 2)
+    local width_padding = math.floor((max_columns - width) / 2) + bs + 1
+    results.col, preview.col, prompt.col = width_padding, width_padding, width_padding
+
+    local anchor = layout_config.anchor or ""
+    local anchor_pos = resolve.resolve_anchor_pos(anchor, width, height, max_columns, max_lines)
+    adjust_pos(anchor_pos, prompt, results, preview)
+
+    -- Vertical anchoring (S or N variations) ignores layout_config.mirror
+    anchor = anchor:upper()
+    local mirror
+    if anchor:find "S" then
+      mirror = false
+    elseif anchor:find "N" then
+      mirror = true
     else
-      preview.line = topline + (results.height + (2 * bs + 2))
-      preview.height = max_lines - preview.line + (1 - bs)
+      mirror = layout_config.mirror
+    end
+
+    -- Set preview position
+    local block_line = math.min(results.line, prompt.line)
+    if not mirror then -- Preview at top
+      preview.line = 1 + bs
+      preview.height = block_line - (2 + 2*bs)
+    else -- Preview at bottom
+      preview.line = block_line + results.height + 2 + 2*bs
```

Comment:
Stylua is complaining about the `2*bs` here, so should add spaces around the `*`